### DISCUSSION TITLE
Why highlightOnAction is of String type?

### DISCUSSION THREAD

**Discussion by lufton** [getgauge/taiko#2583] [General]
It's confusing, but `setConfig` method expects `highlightOnAction` option [to be a string](https://github.com/getgauge/taiko/blob/8b2bb163e195b4d79bf83aa2316f48219861d108/types/taiko/index.d.ts#L76) (one of 'true' or 'false'). What is the reason for that? Why don't just use `boolean`?

**Comment by zabil** [comment]
Taiko is written in javascript and the actual implementation reads the `TAIKO_HIGHLIGHT_ON_ACTION` variable so it ended up using a string to avoid parsing to boolean. 

But the right way is to represent it as a boolean. Please feel free to send a pull request if you a fix in mind, preferably one that is backward compatible works for both string and boolean.

**Reply by lufton** [reply]
Already: https://github.com/getgauge/taiko/pull/2584
But it can't be backward compatible because of this code: https://github.com/getgauge/taiko/blob/8b2bb163e195b4d79bf83aa2316f48219861d108/lib/config.js#L31-L37
